torch/_inductor/fx_passes/post_grad.py

```diff
@@ -653,33 +656,36 @@ def reinplace_scatters(graph):
             assert node.args[0].op == "placeholder"
             mutated_inputs.add(node.args[0])
 
+    def any_use_of_views_after_node(node, shared_view_nodes, copy_node):
+        node_loc = node_order[node]
+        for view in shared_view_nodes:
+            for user in view.users:
+                # Skip all users before node
+                if node_order[user] <= node_loc:
+                    continue
+                # Skip over the copy_ epilogue node that could get reinplaced
+                if copy_node == user:
+                    continue
+                return True
+        return False
+
     def can_inplace(node, mutated_arg):
         if get_node_storage(mutated_arg) is None:
             return False
         shared_view_nodes = storage_to_nodes[get_node_storage(mutated_arg)]
-        if mutated_arg.op == "placeholder":
+        if any(view.op == "placeholder" for view in shared_view_nodes):
```

Comment:
Why is this needed? It's ... plausibly correct to me, but did you find a case where it was needed?

Reply:
Yes, there was a test case that was failing that used some index function which emitted an unsqueeze.

Reply:
What was the exact graph being produced?

Reply:
The gist is

arg0 <- graph input
buf1 = unsquueze arg0
buf2 = indexput buf1
buf3 = squueze buf2
return buf3

arg0 and buf1 share the memory location but we check whether buf1 is placeholder, so we inplace it, however since arg0 shares the same memory location it would be invalid to inplace unless there was a copy at the epilogue as well.

Reply:
Right, that makes sense. I think at some point when discussing this with you last week I thought of this case, but between then and now I forgot about it :P